File: Extras/pytest/tests/test_02_REST.py
import pytest
import requests

def test_isTutorialScene( unium_endpoint ):
  """basic test using the REST interface"""

  response = requests.get( f"{unium_endpoint}/about" )
  assert response.status_code == 200

  data = response.json()
  assert data["Product"] == 'unium'
  assert data["Company"] == 'gwaredd'
  assert data["Scene"] == 'Tutorial'

File: Extras/nunit/Test02_REST.cs
using Newtonsoft.Json.Linq;
using NUnit.Framework;
using System.Net.Http;
using System.Threading.Tasks;

namespace Unium.Test
{
    [TestFixture]
    public class Test02_REST
    {
        [Test]
        public async Task IsTutorialLevelRunning()
        {
            using( var client = new HttpClient() )
            {
                var res = await client.GetAsync( $"{TestConfig.URL}/about" );
                Assert.IsTrue( res.IsSuccessStatusCode );

                var content = await res.Content.ReadAsStringAsync();
                dynamic about = JToken.Parse( content );

                Assert.IsNotNull( about );
                Assert.AreEqual( "gwaredd",  (string) about.Company );
                Assert.AreEqual( "unium",    (string) about.Product );
                Assert.AreEqual( "Tutorial", (string) about.Scene );
            }
        }
    }
}


